config/initializers/omniauth.rb

```
@@ -43,14 +41,44 @@
   end
 end
 
-Rails.application.config.middleware.use OmniAuth::Builder do
-  provider(:find_developer,
-           name: "find-developer",
-           fields: %i[uid email],
-           uid_field: :uid,
-           path_prefix: "/auth",
-           callback_path: "/auth/find-developer/callback")
-  on_failure do |env|
-    Find::Authentication::SessionsController.action(:failure).call(env)
+if Settings.one_login.enabled
+  # Generate callback URL for any environment
+  host = URI(Settings.find_url).host
+  port = Rails.env.development? && "3001"
+  path = "/auth/one-login/callback"
+  one_login_redirect_uri = if Rails.env.local?
+                             URI::HTTP.build(host:, port:, path:)
+                           else
+                             URI::HTTPS.build(host:, port:, path:)
+                           end
+
+  # Load private key from environment variable
+  begin
+    private_key = OpenSSL::PKey::RSA.new(Settings.one_login.private_key.gsub('\n', "\n"))
+  rescue StandardError => e
+    Rails.logger.error(e)
```

Not sure if this is an issue, but if parsing fails, you log and continue, leaving private_key potentially nil. It’d be safer to re-raise so the app fails fast on misconfiguration